### PR TITLE
Modify S2365: Fix code examples

### PR DESCRIPTION
**Pull request by sebastien-marichal**
See https://github.com/SonarSource/sonar-dotnet/issues/9254

### CODE REVIEW

File: rules/S2365/csharp/rule.adoc [hunk 1]
```diff
@@ -9,11 +9,21 @@ include::../why-dotnet.adoc[]
 [source,csharp,diff-id=1,diff-type=noncompliant]
 ----
 private List<string> _foo = new List<string> { "a", "b", "c" };
```
Comment (CristianAmbrosini) [inline rules/S2365/csharp/rule.adoc:11]: (optional) What about adding using statements as well? Specifically the System.Linq
```cs
using System.Collections.Generic;
using System.Linq;
```

File: rules/S2365/csharp/rule.adoc [hunk 2]
```diff
@@ -23,9 +33,16 @@ public IEnumerable<string> Foo  // Noncompliant: expensive ToList call
 [source,csharp,diff-id=1,diff-type=compliant]
 ----
 private List<string> _foo = new List<string> { "a", "b", "c" };
+private string[] _bar = new string[] { "a", "b", "c" };
```
Comment (CristianAmbrosini) [inline rules/S2365/csharp/rule.adoc:36]: I would take this opportunity to change the field names as well, they should not begin with an underscore IMO

File: rules/S2365/vbnet/rule.adoc [hunk 3]
```diff
@@ -8,11 +8,18 @@ include::../why-dotnet.adoc[]
 
 [source,vbnet,diff-id=1,diff-type=noncompliant]
 ----
-Dim foo = {"a", "b", "c" }
-Property Foo() As String() ' Noncompliant
+Private _foo As New List(Of String) From {"a", "b", "c"}
```
Comment (CristianAmbrosini) [inline rules/S2365/vbnet/rule.adoc:11]: Field names

**Review by CristianAmbrosini — CHANGES_REQUESTED** [review]

File: rules/S2365/csharp/rule.adoc (hunk 1 above)
Comment (sebastien-marichal) [reply rules/S2365/csharp/rule.adoc:11]: The example context is implicitly set to a class.
Adding usings would be incorrect. I would rather not add them.

**Review by CristianAmbrosini — APPROVED** [review]
LGTM!

**Pull request merged**